l10n_us_product/models/product_template.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    weight_oz = fields.Float(
+        compute='_compute_weight_oz',
+        inverse='_inverse_weight_oz',
+        help='Weight of the product, in ounces.',
+    )
+    volume_in = fields.Float(
+        compute='_compute_volume_in',
+        inverse='_inverse_volume_in',
+        help='Volume of the product, in inches cubed.',
+    )
+
+    @api.multi
+    def _compute_weight_oz(self):
+        for record in self:
+            record.weight_oz = self._convert_kilogram_ounce(
+                record.weight,
+            )
+
+    @api.multi
+    def _inverse_weight_oz(self):
+        for record in self:
+            record.weight = self._convert_kilogram_ounce(
+                record.weight_oz, to_oz=False,
+            )
+
+    @api.multi
+    def _compute_volume_in(self):
+        for record in self:
+            record.volume_in = self._convert_meter_inch(record.volume)
+
+    @api.multi
+    def _inverse_volume_in(self):
+        for record in self:
+            record.volume = self._convert_meter_inch(
+                record.volume_in, to_in=False,
+            )
+
+    @api.model_cr_context
+    def _convert_kilogram_ounce(self, quantity, to_oz=True):
```

Comment:
Note this is overly complex to allow for usage in other models (such as the module I'm about to submit for stock.picking)